ui/ui-components/pages/app/[id]/edit.tsx

```diff
@@ -211,54 +269,66 @@ export default function Page(props) {
                             if (options[opt.key]?.type === "typeahead") {
                               return (
                                 <>
-                                  <Typeahead
-                                    id="typeahead"
-                                    labelKey="key"
-                                    disabled={loading}
-                                    onChange={(selected) => {
-                                      updateOption(opt.key, selected[0]?.key);
-                                    }}
-                                    options={options[opt.key]?.options.map(
-                                      (k, idx) => {
-                                        return {
-                                          key: k,
-                                          descriptions:
-                                            options[k]?.descriptions[idx],
-                                        };
-                                      }
-                                    )}
-                                    placeholder={
-                                      opt.placeholder || `Select ${opt.key}`
-                                    }
-                                    renderMenuItemChildren={(
-                                      opt,
-                                      props,
-                                      idx
-                                    ) => {
-                                      return [
-                                        <span key={`opt-${idx}-key`}>
-                                          {opt.key}
-                                          <br />
-                                        </span>,
-                                        <small
-                                          key={`opt-${idx}-descriptions`}
-                                          className="text-small"
-                                        >
-                                          <em>
-                                            Descriptions:{" "}
-                                            {opt.descriptions
-                                              ? opt.descriptions.join(", ")
-                                              : "None"}
-                                          </em>
-                                        </small>,
-                                      ];
+                                  <Controller
```

Comment:
Controller component for Typeahead integration or react-hook-form didn't work right 🙃